src/components/TwoFactorAuthModal.js

```diff
@@ -0,0 +1,176 @@
+    secret: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.number,
+    ]),
+  };
+
+  static defaultProps = {
+    modal: {},
+    secret: '',
+    onTwoFactorAuthSubmit: () => ({}),
+  };
+
+
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.modal.isTwoFactorAuthModalOpen !== prevState.isInfoModalOpen) {
+      return {
+        isInfoModalOpen: nextProps.modal.isTwoFactorAuthModalOpen
+      };
+    }
+    return null;
+  }
+
+  onChangeAutoSubmit = e => {
+    const { firstAutoSubmit } = this.state;
+    const currentCode = e.target.value;
+    if (currentCode.length <= 6) {
+      this.setState({currentCode, codeIsWrong: false});
+    } else {
+      this.setState({ currentCode: currentCode.slice(0, 6) });
+    }
+    if (currentCode.length === 6 && firstAutoSubmit) {
+      this.submitForm();
+    }
+  };
+
+  submitForm = async () => {
+    const {
+      onTwoFactorAuthSubmit,
+      closeTwoFactorAuthModalWindow,
+      modal: {mode},
+      disableTFA,
+      confirmTFA
+    } = this.props;
+    const { currentCode } = this.state;
+    try {
+      mode === 'disable' ?
+        await disableTFA(currentCode) :
+        await confirmTFA(currentCode);
+      onTwoFactorAuthSubmit(currentCode);
+      closeTwoFactorAuthModalWindow();
+    } catch(e) {
+      this.setState({codeIsWrong: true, firstAutoSubmit: false});
+    }
+  };
+
+  onEnterPress = e => {
+    const {currentCode} = this.state;
+    if (e.keyCode === 13 && e.shiftKey === false && currentCode.length === 6) {
+      e.preventDefault();
+      this.submitForm();
+    }
+  }
+  
+  onPaste = e => {
+    const pasteValue = e.clipboardData.getData('Text');
+    if ((/^[0-9]{6}$/i).test(pasteValue)) {
+      this.setState({currentCode: pasteValue.toString()}, ()=>{
+        this.submitForm();
+      });
+    }
+  }
+
+  renderModal = () => {
+    const { isInfoModalOpen, codeIsWrong, currentCode } = this.state;
+    const { closeTwoFactorAuthModalWindow,  modal: {mode, authData } } = this.props;
+    const { username, secret } = mode === 'enable' && authData;
+    return (
+      <ModalWindow
+        modalIsOpen={isInfoModalOpen}
+        onClose={closeTwoFactorAuthModalWindow}
+        title={
+          mode === 'enable' ?
+            <FormattedMessage
+              id="saveAndConfirmSecretCode"
+              defaultMessage="Save and confirm secret code"
+            />
+            :
+            <FormattedMessage
+              id="enterConfirmationCode"
+              defaultMessage="Enter confirmation code"
+            />
+        }
+        content={
+          <div className="modal__content-wrapper">
+            {mode === 'enable' &&
+            <div className="modal__secret-wrapper">
+              <div className="modal__qr-wrapper">
+                <QRCode
+                  level="L"
+                  value={`otpauth://totp/Membrana:${username}@membrana.io?secret=${secret}=Membrana`}
```

Comment:
Values `Membrana` and `membrana.io` should be received through props.

Reply:
+